commands/users/banUserMentioned.js

```diff
@@ -0,0 +1,62 @@
+const { SlashCommandBuilder } = require('@discordjs/builders');
+const { guildId, BATId, BALId, BAGId, BADId } = require('../../config.json');
+const { ChannelType } = require('discord-api-types/v9');
+const { MessageAttachment } = require('discord.js');
+
+const ALL_GUILD_IDS = [guildId, BATId, BALId, BAGId, BADId];
+
+module.exports = {
+    data: new SlashCommandBuilder()
+        .setName('banuserinmessage')
+        .setDescription('Gets a list of user ids mentioned in a message(s) and bans them from all servers.')
+        .addStringOption(option =>
+            option.setName('message_ids')
+                .setDescription('The message id(s) you would like to get the user ids from')
+                .setRequired(true))
+        .addChannelOption(option =>
+            option.setName('channel')
+                .setDescription('The channel the message(s) are in')
+                .addChannelTypes([ChannelType.GuildText, ChannelType.GuildPublicThread, ChannelType.GuildPrivateThread])
+                .setRequired(true))
+        .setDefaultPermission(false),
+    async execute(interaction) {
+        const options = interaction.options;
+        const messageIds = options.getString('message_ids');
+        const channel = options.getChannel('channel');
+
+        await interaction.deferReply();
+
+        const banUsersOnServers = async (messageId) => {
+            channel.messages.fetch(messageId).then(async (msg) => {
+                let membersBanned = '';
+                const userIds = msg.mentions.users.map(user => user.id);
+                for (const id of userIds) {
+                    let errorCount = 0;
+                    for (const idGuild of ALL_GUILD_IDS) {
+                        const guild = interaction.client.guilds.cache.get(idGuild);
+                        if (idGuild && !guild) {
+                            return interaction.followUp({ content: `I can't find server with ID ${idGuild} <a:shookysad:949689086665437184>` });
+                        }
+                        await guild.members.ban(id)
+                            .catch((error) => {
+                                console.log(error);
+                                errorCount += 1;
+                                interaction.followUp({ content: `There was an error banning user <@${id}> in server with ID ${idGuild} <a:shookysad:949689086665437184>` });
+                            });
```

Comment:
wondering if there is a good case here to make this and the kick version of this into a shared method in `discordutil.js` and then reference where it is used since it is shared across multiple files? you'd have to send back failures so you could update that tracking value. 🤔

Reply:
i would love that, it hurt my little heart to copy so much code but my pathetic js skills where making me too scared to try to do that 😆 i was also wondering, if it wouldnt be better to take out lines 36-39 and do that way sooner before starting to loop through messages, so if there's a problem with finding the servers for whatever reason, you wouldnt get thousands of those errors? what do you say? also i will try to extract this to the `discordutil.js` i will most probably need to google how to send back the failures so if i get lost i hope you dont mind me coming for help to you 🥺

Reply:
hehe yes moving 36-39 out of the loop feels great while a bunch of 'I can't find server X' messages feels 😢 

and if moving to `discordutil` isn't super straight forward, i wouldn't worry about it for now and we can keep thinking about how to improve it in the future.